cachi2/core/package_managers/pip.py

```diff
@@ -1552,7 +1558,7 @@ def _download_pypi_package(requirement, pip_deps_dir, pypi_url, pypi_auth=None):
     }
 
 
-def _process_package_links(links, name, version):
+def _process_package_links(links: Iterable, name: str, version: str) -> list[dict[str, str]]:
```

Comment:
Maybe this could be further specified as `Iterable[bs4.Tag]`? (an iterable of beautifulsoup tags)